tools/jenkins/run_interop.sh

```diff
@@ -37,3 +37,5 @@ export LANG=en_US.UTF-8
 cd $(dirname $0)/../..
 
 tools/run_tests/run_interop_tests.py -l all -s all --cloud_to_prod --cloud_to_prod_auth --use_docker --http2_interop -t -j 12 $@ || true
+tools/run_tests/run_interop_tests.py -l java --use_docker --http2_badserver_interop $@ || true
```

Comment:
I was going to click the "HTML Report" link for this test run to see if it contains results from both runs, but this test failed to build docker image: https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/10846/console.

Also currently it probably won't show results for the 2nd test because the test cases are not added to https://github.com/grpc/grpc/blob/master/tools/run_tests/run_interop_tests.py#L995. 

One fix for this would be combining the two commands into:
tools/run_tests/run_interop_tests.py -l all -s all --cloud_to_prod --cloud_to_prod_auth --use_docker --http2_interop **--http2_badserver_interop**

In the test script if --http2_badserver_interop is true, automatically run all languages that support this case, currently java only, i.e., ignore -l arg for http2_badserver_interop cases. Then it'll be easy to add this case to HTML. I can take care of that.